gramex/services/__init__.py

```diff
@@ -902,3 +904,38 @@ def test(conf):
     # Remove auth: section when running gramex.
     # If there are passwords here, they will not be loaded in memory
     conf.pop('auth', None)
+
+
+def gramexlog(conf):
+    '''
+    '''
+    from elasticsearch import Elasticsearch
+    from elasticsearch import helpers
+    default_conf = conf.get('default')
+    info.gramexlog.conf = conf
+    info.gramexlog.poll = poll = default_conf.get('poll', 1)
+    info.gramexlog.queue = queue = []
+    info.gramexlog.maxlength = default_conf.get('maxlength', 100000)
+    info.gramexlog.connection = connection = Elasticsearch(
+        default_conf.get('host') or None, http_auth=(
+            default_conf.get('user'), default_conf.get('pass')))
+
+    def log_to_es():
+        try:
+            if queue:
+                # Override _id -- a must for bulk indexing in ElasticSearch
+                for item in queue:
+                    item['_id'] = uuid.uuid4()
+                helpers.bulk(connection, queue)
```

Comment:
Select the connection based on the `_index` in the queue. This means you also need to know which host, user ID, password, etc.